parts/k8s/cloud-init/artifacts/cse_customcloud.sh

```diff
@@ -195,7 +195,12 @@ configureAzureStackInterfaces() {
 
   mapfile -t local_interfaces < <(cat /sys/class/net/*/address | tr -d : | sed 's/.*/\U&/g')
 
-  SDN_INTERFACES=$(jq ".value | map(select(.properties.macAddress | inside(\"${local_interfaces[*]}\"))) | map(select((.properties.ipConfigurations | length) > 0))" ${NETWORK_INTERFACES_FILE})
+  SDN_INTERFACES=$(jq ".value | map(select(.properties != null) | select(.properties.macAddress != null) | select(.properties.macAddress | inside(\"${local_interfaces[*]}\"))) | map(select((.properties.ipConfigurations | length) > 0))" ${NETWORK_INTERFACES_FILE})
```

Comment:
How is this additional null protection actually fixing anything?

Reply:
there is an underlying issue on upgrade where the upgrade for Windows nodes is creating an extra nic. The nic is not attached to a VM thus it doesnt have a macAddress property. Without this null check for macAddress jq throws an error which prevents the configuration file to be created

Reply:
So basically this change filters out all interfaces that have a null properties or properties.macAddress value, and in practice the right interface is selected?

How are we addressing the underlying issue of an unexpected additional NIC?

Reply:
This change is a defensive fix to unblock the CNI scenario for Azure Stack on upgrade, I am still investigating the issue for the additional NIC, will send a separate PR for it once I've figured it out.